config_types.cpp

```diff
@@ -263,3 +266,93 @@ const char* server_addr::get_last_error(void) const
 {
     return gai_strerror(m_last_error);
 }
+
+config_cpu_list::config_cpu_list(unsigned int cores)
+{
+    for (unsigned int i = 0; i < cores; ++i)
+        cpu_list.push_back(i);
+
+    next_cpu = cpu_list.begin();
+}
+
+config_cpu_list::config_cpu_list(std::string str)
+{
+    std::stringstream ss(str);
+
+    if (str.find(' ') != std::string::npos)
+        return;
+
+    char separator = str[str.find_first_of(",-")];
```

Comment:
following my previous question - how do you deal with both `,` and `-` seperators?

Reply:
see above